pkg/providers/amifamily/types.go

```diff
@@ -99,12 +99,13 @@ type DescribeImageQuery struct {
 	KnownRequirements map[string][]scheduling.Requirements
 }
 
-func (q DescribeImageQuery) DescribeImagesInput() *ec2.DescribeImagesInput {
+func (q DescribeImageQuery) DescribeImagesInput(includeDeprecated bool) *ec2.DescribeImagesInput {
 	return &ec2.DescribeImagesInput{
 		// Don't include filters in the Describe Images call as EC2 API doesn't allow empty filters.
-		Filters:    lo.Ternary(len(q.Filters) > 0, q.Filters, nil),
-		Owners:     lo.Ternary(len(q.Owners) > 0, lo.ToSlicePtr(q.Owners), nil),
-		MaxResults: aws.Int64(1000),
+		Filters:           lo.Ternary(len(q.Filters) > 0, q.Filters, nil),
+		Owners:            lo.Ternary(len(q.Owners) > 0, lo.ToSlicePtr(q.Owners), nil),
+		IncludeDeprecated: lo.ToPtr(includeDeprecated),
```

Comment:
Quick check: Does changing this from not specified to false change any behavior?

Reply:
Nope, if you're the AMI ownder, deprecated AMIs are included regardless:

```go
// service/ec2/api.go:97975

	// Specifies whether to include deprecated AMIs.
	//
	// Default: No deprecated AMIs are included in the response.
	//
	// If you are the AMI owner, all deprecated AMIs appear in the response regardless
	// of what you specify for this parameter.
	IncludeDeprecated *bool `type:"boolean"`
```